### PR TITLE
feat: enable filtering in delegated routing client

### PR DESCRIPTION
**Pull request by 2color**
## Description

- **deps: upgrade delegated routing client**
- **feat: leverage filtering for delegated routing in libp2p constructor** 

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/helia/src/utils/libp2p-defaults.browser.ts [hunk 1]
```diff
@@ -64,7 +64,10 @@ export function libp2pDefaults (options: Libp2pDefaultsOptions = {}): Libp2pOpti
     services: {
       autoNAT: autoNAT(),
       dcutr: dcutr(),
-      delegatedRouting: () => createDelegatedRoutingV1HttpApiClient('https://delegated-ipfs.dev'),
+      delegatedRouting: () => createDelegatedRoutingV1HttpApiClient('https://delegated-ipfs.dev', {
+        filterProtocols: ['unknown', 'transport-bitswap', 'transport-ipfs-gateway-http'],
```
Comment (SgtPooki) [inline packages/helia/src/utils/libp2p-defaults.browser.ts:68]: what does the `unknown` filter protocol give us here? will that increase unreachable peers?

File: packages/helia/src/utils/libp2p-defaults.ts [hunk 2]
```diff
@@ -73,7 +73,10 @@ export function libp2pDefaults (options: Libp2pDefaultsOptions = {}): Libp2pOpti
     services: {
       autoNAT: autoNAT(),
       dcutr: dcutr(),
-      delegatedRouting: () => createDelegatedRoutingV1HttpApiClient('https://delegated-ipfs.dev'),
+      delegatedRouting: () => createDelegatedRoutingV1HttpApiClient('https://delegated-ipfs.dev', {
+        filterProtocols: ['unknown', 'transport-bitswap', 'transport-ipfs-gateway-http'],
```
Comment (SgtPooki) [inline packages/helia/src/utils/libp2p-defaults.ts:77]: ditto for `unknown`

File: packages/helia/src/utils/libp2p-defaults.browser.ts (hunk 1 above)
Comment (2color) [reply packages/helia/src/utils/libp2p-defaults.browser.ts:68]: See https://github.com/ipfs/specs/pull/484/files#diff-2f9905a73a1926c9149e7bab39df73763be227e2d6f54fd16b708ee605369905R69

TL;DR: it's to include DHT providers, whose protocol isn't explicitly `bitswap` (even though in almost all cases they support bitswap)

File: packages/helia/src/utils/libp2p-defaults.ts [hunk 3]
```diff
@@ -73,7 +73,10 @@ export function libp2pDefaults (options: Libp2pDefaultsOptions = {}): Libp2pOpti
...
+        filterAddrs: ['https', 'tcp', 'webrtc', 'webrtc-direct', 'wss']
```
Comment (SgtPooki) [inline packages/helia/src/utils/libp2p-defaults.ts:78]: we probably want to include `ws` & `http` here as well?

Comment (achingbrain) [reply packages/helia/src/utils/libp2p-defaults.ts:78]: I can see that would be useful in test environments but probably not by default?

**Review by achingbrain — APPROVED** [review]
